mtupper/homework.md

```diff
@@ -0,0 +1,292 @@
+>```
+
+**l)** `x++ == ++x`
+> **false** - value of x is returned and then incremented and incremented again before being compared to the new value of x
+>
+> ```javascript
+> var x = 1;
+> var y = x++; // y = 1, x = 2
+> var z = ++x; // z = 3, x = 3
+> y == z; // is false
+> ```
+
+**m)** `++x == x++`
+> **true** - value of x is incremented and returned, new value of x is compared to x (true), then incremented.
+>
+> ```javascript
+> var x = 1;
+> var y = ++x; // y = 2, x = 2;
+> var z = x++; // z = 2, x = 3;
+> y == z; // is true
+> ```
+
+**n)** `"1"+x == 1+x`
+> **false** - treats addition of strings different from addition of integers
+>
+> ```javascript
+> var x = 4;
+> "1" + x; // returns "14"
+>
+> var x = 4;
+> 1 + x; // returns 5
+>
+> "14" == 5; // returns false
+> ```
+
+**o)** `"0"+1 == 1`
+> **true** - will convert strings to numbers for comparison with non-strict (==) equality
+> ```javascript
+> "0" + 1; // returns "01"
+>
+> "01" == 1; // returns true
+> ```
+
+**p)** `(typeof (x+1))===(typeof x)`
+> **true** - comparing `"numbers"`
+> ```javascript
+> var x = 4;
+> (typeof (x + 1)); // is same as (5) and returns "number"
+> (typeof x); // is same as 4 and also returns "number"
+> ```
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+> **true** - since `(x*1 == x)` evaluates to true, second statement is optional (or `||`), it returns true without needing to evaluate the second statement
+
+**r)** `(x=(typeof (x+(typeof x))))==x`
+> **true** - the statement on the left with always result in `"string"` first and then compare it to itself
+
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+ **2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.<br />
+> ```javascript
+> var avg = (x+y+z)/3;
+> ```
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+> ```javascript
+> var avg = (x+y+z)/3,
+>     x += (avg-x)/2,
+>     y += (avg-y)/2,
+>     z += (avg-z)/2;
+> ```
+
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+
+> ```javascript
+> var w = (r - l),
+>     h = (t - b),
+>     a = w * h;
+> ```
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+> ```javascript
+> var w = (r - l),
+>     h = (t - b);
+>
+> if ( h > w );
+> ```
```

Comment:
If you are going toput it in an if statment you need to finish it. ie. if(h > w) { ans = true} else {ans = false}. But i get what you are going at there.